lib/codecs/src/decoding/format/json.rs

```diff
@@ -16,7 +18,34 @@ use super::Deserializer;
 
 /// Config used to build a `JsonDeserializer`.
 #[derive(Debug, Clone, Default, Deserialize, Serialize)]
-pub struct JsonDeserializerConfig;
+pub struct JsonDeserializerConfig {
+    #[serde(
+        default,
+        skip_serializing_if = "vector_core::serde::skip_serializing_if_default"
+    )]
+    /// Options for the JSON deserializer.
+    pub json: JsonDeserializerOptions,
+}
+
+/// Options for building a `JsonDeserializer`.
```

Comment:
Since this gets shoved into the website documentation, we generally want to use more user-facing language rather than trying it like an internal code comment.

Something like `JSON-specific decoding options`, which matches the format of other encoder/decoder codec subconfigs.